src/wwwroot/app/views/reports/users-report/users.report.controller.js

```diff
@@ -136,21 +161,88 @@ export default function UsersReportController(
    function clear() {
       _clearLocationField();
       _clearUserField();
-      vm.usersReportParametrs              = {};
-      vm.reportGroupedByLocation           = {};
-      vm.reportGroupedByUser               = {};
-      vm.usersReportParametrs.locationIds  = [];
-      vm.usersReportParametrs.userIds      = [];
+      vm.usersReportParametrs                = {};
+      vm.reportGroupedByLocation             = {};
+      vm.reportGroupedByUser                 = {};
+      vm.reportGroupedByLocationForHistogram = {};
+      vm.reportGroupedByUserForHistogram     = {};
+      vm.usersReportParametrs.locationIds    = [];
+      vm.usersReportParametrs.userIds        = [];
+      //remove report chart on clean report parametrs
+      vm.$broadcast('onClear');
    }
 
    function _convertReport(report) {
       if (vm.selectedLocations.length) {
          set(vm, 'reportGroupedByLocation', _convertReportToHash('locationId', report));
+         set(vm, 'reportGroupedByLocationForHistogram', reverse(_converReportForHistogram('locationId',
+            report)));
       } else {
          set(vm, 'reportGroupedByUser', _convertReportToHash('userId', report));
+         set(vm, 'reportGroupedByUserForHistogram', reverse(_converReportForHistogram('userId',
+            report)));
       }
    }
 
+   function _converReportForHistogram(key, value) {
+      let userGroupedByStagesObj = groupBy(_flattenResponseObgect(value), 'stageTitle');
+      if (key === 'locationId') {
+         return reduce(userGroupedByStagesObj, (histogramArr, stageGroup, stageTitle) => {
+            let groupedByLocationObj = groupBy(stageGroup, 'location');
+            let mappedArr = map(groupedByLocationObj, (locationGroup, locationTitle) => {
+               return {
+                  name: locationTitle,
+                  value: sumBy(locationGroup, (userValue) => {
+                     return userValue.number;
+                  }),
+                  users: sortBy(locationGroup, [ 'number' ])
+               };
+            });
+            let resultHistogramObgect = {
+               state: stageTitle,
+               values: mappedArr
+            };
+            histogramArr.push(resultHistogramObgect);
+            return histogramArr;
+         }, []);
+      } else {
+         return reduce(userGroupedByStagesObj, (histogramArr, stageGroup, stageTitle) => {
+            let userObjArray = map(stageGroup, (groupElement) => {
+               return {
+                  name: groupElement.name,
+                  value: groupElement.number
+               };
+            });
+            let resultHistogramObgect = {
+               state: stageTitle,
+               values: sortBy(userObjArray, [ 'value' ])
+            };
+            histogramArr.push(resultHistogramObgect);
```

Comment:
you can use **map** instead of **reduce**. just return **resultHistogramObgect**